### PR TITLE
NUI-937 add Githbub Action for unit tests and semantic release

### PR DESCRIPTION
**Pull request by jdelbick**
## Description

release steps are a little different for devtool: https://github.com/adobe/asset-compute-devtool#releasing
release require `npm run build` in the `server` folder

Add tests & semantic release to github action workflow.

Inspired by helix team's [semantic release github action](https://github.com/adobe/htlengine/blob/main/.github/workflows/semantic-release.yaml) and the open source teams example [github action](https://github.com/adobe/starter-repo/blob/master/.github/workflow/npm-publish.yml)

## Changes
- add `client-test` and `server-test` (which also includes `npm run build` to build the static ui)
- add step for semantic release
- add "skip" conditions to each step that follow our normal semantic release `[ci skip]` [messages]](https://github.com/adobe/aio-cli-plugin-asset-compute/blob/master/.releaserc.json#L31)
- add semantic release library and config (this repo did not have it before)

Fixes # https://jira.corp.adobe.com/browse/NUI-937

## Testing
Tested on a dummy repo of my own & then successfully on one of our nui repos: [node-metrics-sampler](https://github.com/adobe/node-metrics-sampler)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by jdelbick** [comment]
fyi I only added build & release for the first iteration, not tests

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js Test CI
```
Comment (tmathern) [inline .github/workflows/release.yml:4]:
```suggestion
name: Node.js CI
```

**Comment by jdelbick** [comment]
we will call this one a WIP right now while I figure out a way to verify if this code will work^^^ @tmathern

**Comment by jdelbick** [comment]
I got the [build & tests](https://github.com/adobe/asset-compute-devtool/runs/1500656154?check_suite_focus=true#step:4:468) to run (build is the extra step needed for releasing devtool), so the rest should work the same as the rest of the repos

**Review by tmathern — APPROVED** [review]

**Pull request merged**